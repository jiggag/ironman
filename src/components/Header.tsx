import React, { memo, useCallback } from 'react';
import { View } from 'react-native-ui-lib';
import { Text, TouchableOpacity, StyleSheet } from 'react-native';
import { useSelector } from 'react-redux';
import { useNavigation } from '@react-navigation/native';
import Constant from '../utils/constants';
import HeaderRightButton from './HeaderRightButton';

interface HeaderType {
  onPress: () => void;
  onPressRightButton?: () => void;
  onPressDelete?: () => void;
  type?: string;
}

const TYPE = {
  CREATE: '생성',
  UPDATE: '수정',
  SAVE: '저장',
  SEND: '보내기'
};

const Header = memo(({
  onPress, onPressRightButton, onPressDelete, type,
}: HeaderType) => {
  const navigation = useNavigation();
  const { auth } = useSelector(store => store.user);
  const onPressVoc = useCallback(() => {
    navigation.navigate(auth === "ADMIN" ? "ListVoc" : "SendVoc")
  }, []);
  
  return (
    <View row style={styles.header}>
      <View flex left>
        <TouchableOpacity activeOpacity={0.8} onPress={onPress}>
          <View paddingV-2>
            <Text style={styles.backButtonText}>뒤로가기</Text>
          </View>
        </TouchableOpacity>
      </View>
      <HeaderRightButton
        isVisible
        buttonIdx={2}
        text="문의"
        onPress={onPressVoc}
        style={styles.deleteButtonText}
      />
      <HeaderRightButton
        isVisible={type === 'UPDATE'}
        buttonIdx={1}
        text="삭제"
        onPress={onPressDelete}
        style={styles.deleteButtonText}
      />
      <HeaderRightButton
        isVisible={!!type}
        buttonIdx={0}
        text={TYPE[type]}
        onPress={onPressRightButton}
        style={styles.backButtonText}
      />
    </View>
  )
});

export default Header;

const styles = StyleSheet.create({
  header: {
    backgroundColor: Constant.WHITE_COLOR,
    paddingVertical: 8,
    paddingHorizontal: 10,
    elevation: 5,
    shadowColor: Constant.SHADOW_COLOR,
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.1,
    shadowRadius: 1,
    marginBottom: 2,
  },
  backButtonText: {
    color: Constant.SHADOW_COLOR,
    fontWeight: '600',
    fontSize: 14,
  },
  deleteButtonText: {
    color: Constant.WARN_COLOR,
    fontWeight: '600',
    fontSize: 14,
  },
});
