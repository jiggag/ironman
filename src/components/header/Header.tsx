import React, { memo, useCallback } from 'react';
import { StyleSheet } from 'react-native';
import { View } from 'react-native-ui-lib';
import { useNavigation } from '@react-navigation/native';
import { useSelector } from 'react-redux';
import { BackButton } from '@components/header/BackButton';
import Constant, { FontSize, FontWeight } from '@utils/constants';
import { RootReducer } from '../../types';
import { HeaderRightButton } from './HeaderRightButton';

interface HeaderType {
  onPress: () => void;
  onPressRightButton?: () => void;
  onPressDelete?: () => void;
  type?: string;
}

export enum BUTTON_TYPE {
  CREATE = 'CREATE',
  UPDATE = 'UPDATE',
  SAVE = 'SAVE',
  SEND = 'SEND',
}

const BUTTON_TEXT = {
  CREATE: '생성',
  UPDATE: '수정',
  SAVE: '저장',
  SEND: '보내기',
};

export const Header = memo<HeaderType>(({
  onPress, onPressRightButton, onPressDelete, type,
}) => {
  const { auth } = useSelector((store: RootReducer) => store.user);
  const navigation = useNavigation();

  const onPressVoc = useCallback(() => {
    navigation.navigate(auth === 'ADMIN' ? 'ListVoc' : 'SendVoc');
  }, [auth, navigation]);

  return (
    <View row centerV paddingH-20 style={styles.header}>
      <BackButton onPress={onPress} />
      <HeaderRightButton isVisible buttonIdx={2} text="문의" onPress={onPressVoc} style={styles.deleteButtonText} />
      <HeaderRightButton
        isVisible={type === 'UPDATE'}
        buttonIdx={1}
        text="삭제"
        onPress={onPressDelete}
        style={styles.deleteButtonText}
      />
      <HeaderRightButton
        isVisible={!!type}
        buttonIdx={0}
        text={BUTTON_TEXT[type as string]}
        onPress={onPressRightButton}
        style={styles.buttonText}
      />
    </View>
  );
});

const styles = StyleSheet.create({
  buttonText: {
    color: Constant.SHADOW_COLOR,
    fontSize: FontSize.normal,
    fontWeight: FontWeight.bold,
    paddingHorizontal: 10,
  },
  deleteButtonText: {
    color: Constant.WARN_COLOR,
    fontSize: FontSize.normal,
    fontWeight: FontWeight.bold,
    paddingHorizontal: 10,
  },
  header: {
    backgroundColor: Constant.WHITE_COLOR,
    elevation: 5,
    height: 50,
    marginBottom: 2,
    shadowColor: Constant.SHADOW_COLOR,
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.1,
    shadowRadius: 1,
  },
});
