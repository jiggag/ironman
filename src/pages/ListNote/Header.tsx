import React from 'react';
import { View } from 'react-native-ui-lib';
import PropTypes from 'prop-types';
import { LineGraph } from '../../components';

const Header = ({ data }) => {
  return (
    <View marginV-10>
      <LineGraph data={data} />
    </View>
  );
};

export default Header;

Header.defaultProps = {
  data: [],
};
Header.propTypes = {
  data: PropTypes.array,
};
