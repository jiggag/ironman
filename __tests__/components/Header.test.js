import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../../src/components/Header';

it('Renderer Component: Header', () => {
  const createComponent = renderer.create(<Header onPress={jest.fn()} onPressRightButton={jest.fn()} type="CREATE" />);
  let createHeader = createComponent.toJSON();
  expect(createHeader).toMatchSnapshot();

  const updateComponent = renderer.create(
    <Header onPress={jest.fn()} onPressRightButton={jest.fn()} onPressDelete={jest.fn()} type="UPDATE" />
  );
  let updateHeader = updateComponent.toJSON();
  expect(updateHeader).toMatchSnapshot();
});
