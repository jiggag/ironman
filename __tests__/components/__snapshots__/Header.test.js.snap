// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderer Component: Header 1`] = `
<View
  row={true}
  style={
    Array [
      undefined,
      undefined,
      undefined,
      Object {},
      Object {},
      Object {
        "flexDirection": "row",
      },
      Object {
        "backgroundColor": "#ffffff",
        "elevation": 5,
        "marginBottom": 2,
        "paddingHorizontal": 10,
        "paddingVertical": 8,
        "shadowColor": "#333333",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        undefined,
        undefined,
        Object {
          "flex": 1,
        },
        Object {},
        Object {},
        Object {
          "alignItems": "flex-start",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        paddingV-2={true}
        style={
          Array [
            undefined,
            undefined,
            undefined,
            Object {
              "paddingVertical": 2,
            },
            Object {},
            Object {},
            undefined,
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#333333",
              "fontSize": 14,
              "fontWeight": "600",
            }
          }
        >
          뒤로가기
        </Text>
      </View>
    </View>
  </View>
  <View
    centerV={true}
    marginR-10={true}
    style={
      Array [
        undefined,
        undefined,
        undefined,
        Object {},
        Object {
          "marginRight": 10,
        },
        Object {
          "alignItems": "flex-end",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        paddingH-6={true}
        paddingV-4={true}
        style={
          Array [
            undefined,
            undefined,
            undefined,
            Object {
              "paddingHorizontal": 6,
              "paddingVertical": 4,
            },
            Object {},
            Object {},
            undefined,
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#e34d10",
              "fontSize": 14,
              "fontWeight": "600",
            }
          }
        >
          문의
        </Text>
      </View>
    </View>
  </View>
  <View
    centerV={true}
    marginR-10={false}
    style={
      Array [
        undefined,
        undefined,
        undefined,
        Object {},
        Object {},
        Object {
          "alignItems": "flex-end",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        paddingH-6={true}
        paddingV-4={true}
        style={
          Array [
            undefined,
            undefined,
            undefined,
            Object {
              "paddingHorizontal": 6,
              "paddingVertical": 4,
            },
            Object {},
            Object {},
            undefined,
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#333333",
              "fontSize": 14,
              "fontWeight": "600",
            }
          }
        >
          생성
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Renderer Component: Header 2`] = `
<View
  row={true}
  style={
    Array [
      undefined,
      undefined,
      undefined,
      Object {},
      Object {},
      Object {
        "flexDirection": "row",
      },
      Object {
        "backgroundColor": "#ffffff",
        "elevation": 5,
        "marginBottom": 2,
        "paddingHorizontal": 10,
        "paddingVertical": 8,
        "shadowColor": "#333333",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        undefined,
        undefined,
        Object {
          "flex": 1,
        },
        Object {},
        Object {},
        Object {
          "alignItems": "flex-start",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        paddingV-2={true}
        style={
          Array [
            undefined,
            undefined,
            undefined,
            Object {
              "paddingVertical": 2,
            },
            Object {},
            Object {},
            undefined,
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#333333",
              "fontSize": 14,
              "fontWeight": "600",
            }
          }
        >
          뒤로가기
        </Text>
      </View>
    </View>
  </View>
  <View
    centerV={true}
    marginR-10={true}
    style={
      Array [
        undefined,
        undefined,
        undefined,
        Object {},
        Object {
          "marginRight": 10,
        },
        Object {
          "alignItems": "flex-end",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        paddingH-6={true}
        paddingV-4={true}
        style={
          Array [
            undefined,
            undefined,
            undefined,
            Object {
              "paddingHorizontal": 6,
              "paddingVertical": 4,
            },
            Object {},
            Object {},
            undefined,
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#e34d10",
              "fontSize": 14,
              "fontWeight": "600",
            }
          }
        >
          문의
        </Text>
      </View>
    </View>
  </View>
  <View
    centerV={true}
    marginR-10={true}
    style={
      Array [
        undefined,
        undefined,
        undefined,
        Object {},
        Object {
          "marginRight": 10,
        },
        Object {
          "alignItems": "flex-end",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        paddingH-6={true}
        paddingV-4={true}
        style={
          Array [
            undefined,
            undefined,
            undefined,
            Object {
              "paddingHorizontal": 6,
              "paddingVertical": 4,
            },
            Object {},
            Object {},
            undefined,
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#e34d10",
              "fontSize": 14,
              "fontWeight": "600",
            }
          }
        >
          삭제
        </Text>
      </View>
    </View>
  </View>
  <View
    centerV={true}
    marginR-10={false}
    style={
      Array [
        undefined,
        undefined,
        undefined,
        Object {},
        Object {},
        Object {
          "alignItems": "flex-end",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        paddingH-6={true}
        paddingV-4={true}
        style={
          Array [
            undefined,
            undefined,
            undefined,
            Object {
              "paddingHorizontal": 6,
              "paddingVertical": 4,
            },
            Object {},
            Object {},
            undefined,
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#333333",
              "fontSize": 14,
              "fontWeight": "600",
            }
          }
        >
          수정
        </Text>
      </View>
    </View>
  </View>
</View>
`;
